scene/resources/texture.cpp

```diff
@@ -370,6 +371,15 @@ void PortableCompressedTexture2D::_set_data(const Vector<uint8_t> &p_data) {
 				uint32_t mipsize = decode_uint32(data);
 				data += 4;
 				data_size -= 4;
+
+				// Skip possible Godot png tag "PNG " before pure PNG data.
+				static const uint8_t godot_png_prefix[] = { 0x50, 0x4E, 0x47, 0x20 };
+				static const size_t godot_png_prefix_size = sizeof(godot_png_prefix);
+				if (memcmp(data, godot_png_prefix, godot_png_prefix_size) == 0) {
+					data += godot_png_prefix_size;
+					data_size -= godot_png_prefix_size;
+				}
```

Comment:
How well does this interact with WebP files? I don't see any type checking to ensure that we aren't breaking WebP files in the process.

Reply:
For WebP format, this code is safe: WebP image data starts with `RIFF` characters

I also don't like this kind of interference with the raw data at this point, where it's not yet clear what format this image actually has. I would prefer to know the format first, and then perform such manipulations.

If you know where to move this code, please tell me! I'll be very happy.

Reply:
Its probably a good idea to keep the implementation similar to CompressedTexture2D in that case. CompressedTexture2D stores a ``DataFormat`` enum which is used (among other things) to signal what format the texture is using. That way we would know whether PNG or WEBP is used and can load the information accordingly.

Reply:
I can't do this because `CompressedTexture2D` determines the format by the first byte in the [`load_image_from_file`](https://github.com/godotengine/godot/blob/46424488edc341b65467ee7fd3ac423e4d49ad34/scene/resources/texture.cpp#L657C21-L657C37) method, where file is the `ctex` file created by the `ResourceImporterTexture` class in method [`save_to_ctex_format`](https://github.com/godotengine/godot/blob/46424488edc341b65467ee7fd3ac423e4d49ad34/editor/import/resource_importer_texture.cpp#L249C23-L249C23).

A `PortableCompressedTexture2D` is created not from a file, but from an `Image` instance, which does not contain this information. And this information cannot be added to it, as this will damage so many other places in the code.

If you know how to do this, I will be very glad to your advice!